paddle/fluid/inference/api/paddle_pass_builder.h

```diff
@@ -125,8 +129,8 @@ class CpuPassStrategy : public PassStrategy {
                 "conv_eltwiseadd_bn_fuse_pass",  // preserve correct pass order
                 "conv_bias_mkldnn_fuse_pass",    //
                 "conv3d_bias_mkldnn_fuse_pass",  //
-                "conv_relu_mkldnn_fuse_pass",    //
-                "conv_elementwise_add_mkldnn_fuse_pass"})) {
+                "conv_elementwise_add_mkldnn_fuse_pass",
+                "conv_relu_mkldnn_fuse_pass"})) {
```

Comment:
why do you change the order here?

Reply:
Because it was in the wrong order. It was overlooked in the previous fix of mkldnn passes.

Reply:
Could you please illustrate an example? Thanks.

Reply:
How about move the implementation of `EnableMKLDNN` and `EnableMkldnnQuantizer` to `paddle_pass_builder.cc`?

Reply:
Sure. With ResNet50, after `conv_bias_mkldnn_pass` there picture is as follows:
![image](https://user-images.githubusercontent.com/19307410/55382377-c9d76c00-5525-11e9-9554-8ec81634b4af.png)
Then the `conv_elementwise_add_mkldnn` pass has to be applied:
![image](https://user-images.githubusercontent.com/19307410/55382513-1c188d00-5526-11e9-85cc-34b65435c2ee.png)
and then `conv_relu_mkldnn_fuse_pass`:
![image](https://user-images.githubusercontent.com/19307410/55382551-381c2e80-5526-11e9-87a5-8700a55aa455.png)
In the opposite order we would get remaining relu operators:
![image](https://user-images.githubusercontent.com/19307410/55383155-c5ac4e00-5527-11e9-8549-6153f267686b.png)

It was unnoticable until now, as some passes (`conv_relu_mkldnn_fuse_pass` among them) were unintentionally repeated. Eliminating arbitrary repetitions, the problem becomes visible, and this patch fixes it.

Reply:
Clear now, thanks @wojtuss

Reply:
> How about move the implementation of `EnableMKLDNN` and `EnableMkldnnQuantizer` to `paddle_pass_builder.cc`?

I do not think it can be done in an elegant and user-friendly way. A user has to have an option to enable MKL-DNN and MkldnnQuantizer and provide data necessary for MkldnnQuantizer, so the methods have to be in `AnalysisConfig`. Their implementation there is minimal.

Reply:
I mean could you move line 122-134 from `paddle_pass_builder.h` to `paddle_pass_builder.cc`?

Reply:
Done.